ode-wrapper/src/main/java/com/trihydro/odewrapper/controller/WydotTimIncidentController.java

```diff
@@ -88,6 +91,58 @@ public ResponseEntity<String> createIncidentTim(@RequestBody TimIncidentList tim
         return ResponseEntity.status(HttpStatus.OK).body(responseMessage);
     }
 
+    public void makeIncreasingTims(WydotTimIncident wydotTim) {
+        // i - add buffer for point TIMs
+        WydotTimIncident timOneWay = wydotTim.copy();
+        timOneWay.setDirection("I");
+
+        addTimsToSend(timOneWay, false);
+
+        makeBufferTims(timOneWay);
+    }
+
+    public void makeDecreasingTims(WydotTimIncident wydotTim) {
+        // d - add buffer for point TIMs
+        WydotTimIncident timOneWay = wydotTim.copy();
+        timOneWay.setDirection("D");
+
+        addTimsToSend(timOneWay, false);
+
+        makeBufferTims(timOneWay);
+    }
+
+    private void addTimsToSend(WydotTimIncident tim, boolean isBuffer) {
+        for (String itisCodeEntry : tim.getItisCodes()) {
+            // CTW Update requires that individual TIMs are created for each ITIS ordering
+            List<String> itisCodes = Arrays.asList(itisCodeEntry.split(" "));
+
+            // only generate appropriate TIMs for geometry list (buffer, workZone)
+            String lastItisCode = itisCodes.get(itisCodes.size() - 1);
+            boolean isBufferTim = bufferTimITISCodes.contains(Integer.valueOf(lastItisCode));
+            if (isBuffer && !isBufferTim) {
+                continue;
+            } else if (!isBuffer && isBufferTim) {
+                continue;
+            }
+
+            WydotTimIncident timToSend = tim.copy();
+            timToSend.setItisCodes(itisCodes);
+            var itisCodeAbb = SetItisCodes.getItisCodeAbbreviation(itisCodeEntry);
+            String clientIdWithItis = tim.getClientId() + '-' + tim.getDirection() + '-' + itisCodeAbb;
+            timToSend.setClientId(clientIdWithItis);
+            timsToSend.add(timToSend);
+        }
+    }
+
+    public void makeBufferTims(WydotTimIncident wydotTim) {

```

Comment:
Question (non-blocking): Should we move away from the 'Wydot*' naming convention for our models and instead use 'Dot*'?

Reply:
Eventually, yes. This is outside the scope of this PR though